### PR TITLE
Bump Cluster API to v1.1.1

### PR DESCRIPTION
**Pull request by ykakarap**
### What this PR does / why we need it

This PR updates the version of ClusterAPI to v1.1.1.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by ykakarap** [comment]
//usebom:sandbox/v1.6.0-zshippable/394330997820227902/tkg-compatibility:v5

**Review by maralavi — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -5,13 +5,7 @@ go 1.17
...
 retract [v1.4.0-pre-alpha-1, v1.4.0-pre-alpha-2]
 
-replace (
-	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.5.5 // indirect
-	github.com/k14s/kbld => github.com/anujc25/carvel-kbld v0.31.0-update-vendir
-	sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.0.1
-	sigs.k8s.io/cluster-api/test => sigs.k8s.io/cluster-api/test v1.0.1
-	sigs.k8s.io/kind => sigs.k8s.io/kind v0.11.1
-)
+replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.1.1
```
Comment (rajathagasthya) [inline go.mod:8]: Do we still need this `replace` if `cluster-api` version in L102 is the same? Or is this due to `cluster-api/test` needing it?

**Comment by ykakarap** [comment]
//usebom:sandbox/v1.6.0-zshippable/395233872993896629/tkg-compatibility:v6

**Review by vuil — APPROVED** [review]
lgtm, thanks!

**Pull request merged**